biohansel/create/cluster_generator.py

```diff
@@ -0,0 +1,236 @@
+        for genome, cluster in clusters.iteritems():
+            cluster_genomes[cluster].add(genome)
+        clusters_genomes_dict[threshold] = cluster_genomes
+
+    return clusters_genomes_dict
+
+
+def assign_hc_clusters(clusters_genomes_dict: Dict[float, Dict[int, Set[str]]], min_group_size: int) -> Dict[
+    str, Dict[str, Set[str]]]:
+    """
+    Assigns the subtypes for each genome at each threshold level
+
+    Args:
+        clusters_genomes_dict: A dictionary of the genome cluster groupings with the thresholds used as the key
+        min_group_size: the minimum child group size for each new subtype branching point from the parent group
+
+    Returns:
+        output_clusters: A dictionary within a dictionary that contains each subgtype and the sets of genomes within
+                        that subtype
+    """
+
+    output_subtypes = {threshold: {} for threshold in clusters_genomes_dict.keys()}
+    sorted_thresholds = sorted(clusters_genomes_dict.keys())
+    # initialize top level subtypes
+    output_subtypes[sorted_thresholds[0]] = {str(cluster): genomes for cluster, genomes in
+                                             clusters_genomes_dict[sorted_thresholds[0]].items()}
+
+    for threshold_index in range(1, len(sorted_thresholds)):
+        parent_hc_clusters = output_subtypes[sorted_thresholds[threshold_index - 1]]
+
+        threshold = sorted_thresholds[threshold_index]
+        cluster_genomes = clusters_genomes_dict[threshold]
+
+        for subtype, parent_genomes in parent_hc_clusters.items():
+            if len(parent_genomes) < min_group_size:
+                continue
+            subclade = 1
+            for _, child_genomes in cluster_genomes.items():
+                if len(child_genomes) < min_group_size:
+                    continue
+
+                if parent_genomes == child_genomes:
+                    output_subtypes[threshold][subtype] = parent_genomes
+                elif child_genomes.issubset(parent_genomes):
+                    new_subgroup_name = f'{subtype}.{subclade}'
+                    if len(parent_genomes - child_genomes) >= min_group_size:
+                        output_subtypes[threshold][new_subgroup_name] = child_genomes
+                    else:
+                        output_subtypes[threshold][subtype] = child_genomes
+                    subclade += 1
+
+    return output_subtypes
+
+
+def expand_sets(cluster_dict: Dict[str, Dict[str, Set[str]]]) -> Dict[str, Dict[str, str]]:
+    """
+    Fills in the rest of the cells in the dataframe that had previously been unfilled
+
+    Args:
+        cluster_dict: a dictionary that contains the cluster groupings of each genome with each value being sets of
+                    genomes
+
+    Returns:
+        modified_cluster_dict: A dictionary with values beings filled for all cells
+
+    """
+
+    modified_cluster_dict = {}
+    for threshold, groupings in cluster_dict.items():
+        threshold_dict = {}
```

Comment:
`threshold_dict` -> `genome_cluster`

Also could be dict comprehension:

```python
out[threshold] = {genome: cluster for cluster, genomes in cluster_genomes.items() for genome in genomes}
```

Reply:
addressed in b5b061a66ef9372537e2fcb8891f2f7a747b98db